### PR TITLE
Add auto-publish script

### PR DESCRIPTION
**Pull request by plehegar**
Ideally, we should point to a link where TAG and PING are agreeing to do this. Having said that, this pull request could serve as such.

@torgo @plinss @npdoty @pes10k

### CODE REVIEW

File: .github/workflows/auto-publish.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Configuration options https://w3c.github.io/spec-prod/
+name: Node CI
```
Comment (jyasskin) [inline .github/workflows/auto-publish.yml:2]:
```suggestion
name: Publish the ED and Note
```

File: .github/workflows/auto-publish.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+jobs:
+  validate-and-publish:
+    name: Validate and Publish
+    runs-on: ubuntu-latest # only linux supported at present
+    steps:
+      - uses: actions/checkout@v4
+      - uses: w3c/spec-prod@v2
+        with:
+          TOOLCHAIN: bikeshed
+          W3C_ECHIDNA_TOKEN: ${{ secrets.ECHIDNA_TOKEN }}
```
Comment (jyasskin) [inline .github/workflows/auto-publish.yml:19]: To subsume #169. We should also delete deploy.yml at the same time.

```suggestion
          GH_PAGES_BRANCH: gh-pages
          W3C_ECHIDNA_TOKEN: ${{ secrets.ECHIDNA_TOKEN }}
```

File: .github/workflows/auto-publish.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  push:
+    branches:
+      - main
+  pull_request: {}
+
+jobs:
+  validate-and-publish:
+    name: Validate and Publish
+    runs-on: ubuntu-latest # only linux supported at present
+    steps:
```
Comment (jyasskin) [inline .github/workflows/auto-publish.yml:14]: To allow writing to `gh-pages`:
```suggestion
    permissions:
      contents: write
    steps:
```

**Review by jyasskin — COMMENTED** [review]
A couple changes in order to avoid having two deploy scripts:

**Review by jyasskin — APPROVED** [review]

**Comment by pes10k** [comment]
This looks good to me.  If it looks good to @hober too then i'm good with merging

**Comment by npdoty** [comment]
Privacy Working Group discussed and agreed on this: https://github.com/w3c/privacywg/blob/main/minutes/privacywg-20241121.md#a-securityprivacy-questionnaire

**Comment by pes10k** [comment]
@ylafon , wanted to touch bases with you as follow up here at @hober 's suggestion.  Do you have any thoughts or concerns, or suggestions for further improvement

**Comment by ylafon** [comment]
The SoTD produced by that script has a different licensing requirement section, I don't think there is an issue now, but it was some times ago. If no-one complains about it, I can only be happy that auto-publishing is working now!

**Pull request merged**